routes/users.js

```diff
@@ -1645,15 +1653,18 @@ router.put('/user/:id', async function(req, res) {
         }
 
         user.fid = fid;
-        if(user.oldemail!=user.email && !user.is_fake) {
-            await notif.modify(user.oldemail, user.email);
-        } else if(userWasFake && !user.is_fake) {
-            await notif.add(user.email);
-        }else if (!userWasFake && user.is_fake) {
-            await notif.remove(user.email);
+        // Change mail registration only when user is active
+        // as expired users are removed from mailing list
+        if(user.status == STATUS_ACTIVE) {
```

Comment:
Maybe also allow email change for admin when user is expired

Reply:
it is done on line 1496, here we just prevent mailing list registration (if user not active, should not add him to lists)